pkg/controllers/openshift_apiserver/controller.go

```diff
@@ -0,0 +1,243 @@
+	}
+	configInformers := configinformers.NewSharedInformerFactory(configClient, controllers.DefaultResync)
+	operatorClient := &apiServerOperatorClient{
+		Client:    cfg.KubeClient(),
+		Namespace: cfg.Namespace(),
+		Logger:    cfg.Logger().WithName("OpenShiftAPIServerClient"),
+	}
+
+	recorder := events.NewLoggingEventRecorder("openshift-apiserver-observers")
+	c := configobserver.NewConfigObserver(
+		operatorClient,
+		recorder,
+		configobservation.Listers{
+			ResourceSync:        &noopResourceSyncer{},
+			APIServerLister_:    configInformers.Config().V1().APIServers().Lister(),
+			ImageConfigLister:   configInformers.Config().V1().Images().Lister(),
+			ProjectConfigLister: configInformers.Config().V1().Projects().Lister(),
+			ProxyLister_:        configInformers.Config().V1().Proxies().Lister(),
+			IngressConfigLister: configInformers.Config().V1().Ingresses().Lister(),
+			SecretLister_:       kubeInformers.Core().V1().Secrets().Lister(),
+			PreRunCachesSynced: []cache.InformerSynced{
+				configInformers.Config().V1().Images().Informer().HasSynced,
+				configInformers.Config().V1().Projects().Informer().HasSynced,
+				configInformers.Config().V1().Proxies().Informer().HasSynced,
+				configInformers.Config().V1().Ingresses().Informer().HasSynced,
+			},
+		},
+		images.ObserveInternalRegistryHostname,
+		images.ObserveExternalRegistryHostnames,
+		images.ObserveAllowedRegistriesForImport,
+		ingresses.ObserveIngressDomain,
+		project.ObserveProjectRequestMessage,
+		project.ObserveProjectRequestTemplateName,
+	)
+	cfg.Manager().Add(manager.RunnableFunc(func(stopCh <-chan struct{}) error {
+		configInformers.Start(stopCh)
+		return nil
+	}))
+	cfg.Manager().Add(manager.RunnableFunc(func(stopCh <-chan struct{}) error {
+		ctx, cancel := context.WithCancel(context.Background())
+		go func() {
+			<-stopCh
+			cancel()
+		}()
+		c.Run(ctx, 1)
+		return nil
+	}))
+	configInformers.Config().V1().Images().Informer().AddEventHandler(c.EventHandler())
+	configInformers.Config().V1().Ingresses().Informer().AddEventHandler(c.EventHandler())
+	configInformers.Config().V1().Projects().Informer().AddEventHandler(c.EventHandler())
+	configInformers.Config().V1().Proxies().Informer().AddEventHandler(c.EventHandler())
+	return nil
+}
+
+const (
+	apiserverConfigMapName  = "openshift-apiserver-config"
+	openshiftDeploymentName = "openshift-apiserver"
+)
+
+type apiServerOperatorClient struct {
+	Client    kubeclient.Interface
+	Namespace string
+	Logger    logr.Logger
+}
+
+func (c *apiServerOperatorClient) Informer() cache.SharedIndexInformer {
+	panic("informer not supported")
+}
+
+func (c *apiServerOperatorClient) GetOperatorState() (spec *operatorv1.OperatorSpec, status *operatorv1.OperatorStatus, resourceVersion string, err error) {
+	var cm *corev1.ConfigMap
+	cm, err = c.Client.CoreV1().ConfigMaps(c.Namespace).Get(apiserverConfigMapName, metav1.GetOptions{})
+	if err != nil {
+		return
+	}
+	configYAML := []byte(cm.Data["cofig.yaml"])
```

Comment:
should this be `config.yaml` ?

Reply:
It should, thx! This removes churn in unnecessary updates.